sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs

```diff
@@ -1050,16 +1095,19 @@ private void SetNameFieldsIfNull()
         /// A <see cref="RequestFailedException"/> will be thrown if
         /// a failure occurs.
         /// </remarks>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+#pragma warning disable AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
         public virtual Response<ShareFileCopyInfo> StartCopy(
+#pragma warning restore AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
             Uri sourceUri,
-            Metadata metadata = default,
-            FileSmbProperties smbProperties = default,
-            string filePermission = default,
-            PermissionCopyMode? filePermissionCopyMode = default,
-            bool? ignoreReadOnly = default,
-            bool? setArchiveAttribute = default,
-            ShareFileRequestConditions conditions = default,
-            CancellationToken cancellationToken = default) =>
+            Metadata metadata,
+            FileSmbProperties smbProperties,
+            string filePermission,
+            PermissionCopyMode? filePermissionCopyMode,
+            bool? ignoreReadOnly,
+            bool? setArchiveAttribute,
+            ShareFileRequestConditions conditions,
+            CancellationToken cancellationToken) =>
```

Comment:
Does removing this break the customers that are depending on the default values for any of these parameters if they decide not to specify it?

Reply:
Yes, it will result in a compile time break.  Since default values are a C# feature and not baked into the .NET runtime, you still maintain binary compat.  This is the best option we've found for versioning APIs with default values.